src/providers/auth.service.ts

```diff
@@ -121,4 +122,27 @@ export class AuthService {
       });
     });
   }
+
+  /**
+   * Delete user
+   *
+   * @returns {Observable}
+   */
+  public deleteUser() {
+    return new Observable((observer) => {
+      this.authHttp.delete(environment.server_url + "/api/user")
+        .subscribe(
+          data => {
+            observer.next(data);
+            observer.complete();
+          },
+          err => {
+            observer.next(err);
+            observer.complete();
+          },
+          () => {
+            observer.complete();
```

Comment:
The proper RxJS way of doing the subscription based on their documentation is:   
x => console.log('onNext: %s', x), 
e => console.log('onError: %s', e), 
() => console.log('onCompleted'));

where x is onNext, e is onError and () is onCompleted function handlers